libs/execution_context/ebpf_maps.c

```diff
@@ -813,6 +868,50 @@ ebpf_map_function_table_t ebpf_map_function_tables[] = {
      _next_array_map_key_per_cpu},
 };
 
+static size_t
+_ebpf_core_map_instance_count(ebpf_map_t* map)
+{
+    switch (map->ebpf_map_definition.type) {
+    case BPF_MAP_TYPE_HASH:
+    case BPF_MAP_TYPE_ARRAY:
+    case BPF_MAP_TYPE_PROG_ARRAY:
+        return 1;
+    case BPF_MAP_TYPE_PERCPU_HASH:
+    case BPF_MAP_TYPE_PERCPU_ARRAY:
+        return ((ebpf_core_per_cpu_data_t*)(map->data))->count;
+        break;
+    default:
+        return 1;
+    }
+}
+
+static ebpf_result_t
+_ebpf_reserve_map_space_if_necessary(ebpf_map_t* map, uint32_t cpu_id, const uint8_t* key)
+{
+    bool insert_failed = false;
+    uint32_t index;
+    ebpf_result_t return_value = EBPF_SUCCESS;
+    // check if the space for entry exists
+    if (ebpf_map_function_tables[map->ebpf_map_definition.type].find_entry(map, cpu_id, key) != NULL) {
+        return EBPF_SUCCESS;
+    }
+
+    for (index = 0; index < _ebpf_core_map_instance_count(map); index++) {
+        return_value = ebpf_map_function_tables[map->ebpf_map_definition.type].update_entry(map, index, key, NULL);
```

Comment:
update_entry with zero value should only be done if the entry is not already present on that core.